server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRedirect.java

```diff
@@ -1025,6 +1025,7 @@ protected void updateHeaders() throws MessagingException {
             setIsReply(newMail, isReply(originalMail), originalMail);
 
             newMail.getMessage().saveChanges();
+            newMail.removeAllAttributes();
```

Comment:
Uses : 
 - Bounce
 - DSNBounce
 - Forward
 - Redirect
 - Resend
 - NotifyPostMaster
 - NotifySender

What is a mail attribute  ?

Taken from the JavaDoc of Mail interface : 

--------------
Wraps a MimeMessage with additional routing and processing information.

This includes : 

 - a unique name
[...]
 - additional processing attributes (see below)

[...]

While processing a Mail instance, a Mailet can associate additional
information with it by using mail attributes. These attributes can
then be queried by the same mailet or other mailets later on.
----------

A new mail should then not inherit attributes of the mail it had been generated from. As it has not been processed yet.